File: release/purr-suite-skill-package-2026-03-21/scripts/run_low_stock_scan.ts
import { DEFAULT_DB_PATH, ensureDatabase } from "./db/sqlite.js";
import { emitJson } from "./lib/cli.js";
import { LOW_STOCK_SCAN_DAYS, scanLowStockAlerts } from "./lib/inventory.js";

function parseArgs(argv: string[]): {
  dbPath: string;
  daysAhead: number;
} {
  let dbPath = DEFAULT_DB_PATH;
  let daysAhead = LOW_STOCK_SCAN_DAYS;

  for (let index = 0; index < argv.length; index += 1) {
    const current = argv[index]!;
    if (current === "--db-path") {
      dbPath = argv[index + 1] ?? dbPath;
      index += 1;
      continue;
    }
    if (current === "--days-ahead") {
      const raw = argv[index + 1];
      if (raw) {
        daysAhead = Number.parseInt(raw, 10);
      }
      index += 1;
    }
  }

  return { dbPath, daysAhead };
}

export function main(): number {
  const args = parseArgs(process.argv.slice(2));
  const db = ensureDatabase(args.dbPath);
  try {
    const alerts = db.transaction(() =>
      scanLowStockAlerts(db, {
        mark_sent: true,
        days_ahead: args.daysAhead,
      }),
    )();
    emitJson(
      {
        status: "scanned",
        reply: `Detected ${alerts.length} low-stock alert(s).`,
        alerts,
      },
      0,
    );
    return 0;
  } finally {
    db.close();
  }
}

if (import.meta.url === `file://${process.argv[1]}`) {
  process.exitCode = main();
}

File: Archive/python/scripts/run_low_stock_scan.py
"""Run the low-stock alert scan and emit a stable JSON payload.

This script is a lightweight host/cron entrypoint. It inspects the current
SQLite database, detects low-stock conditions, marks newly pending alerts as
sent, and prints the alert list as stdout JSON.
"""

from __future__ import annotations

import argparse
import json
import sys
from contextlib import closing
from pathlib import Path

ROOT = Path(__file__).resolve().parents[1]
if str(ROOT) not in sys.path:
    sys.path.insert(0, str(ROOT))

from scripts.db.sqlite import DEFAULT_DB_PATH, ensure_database
from scripts.lib.inventory import LOW_STOCK_SCAN_DAYS, scan_low_stock_alerts


def main() -> int:
    """Scan low-stock alerts in the chosen database and print JSON results.

    Args:
        None. Input comes from command-line flags.

    Returns:
        Process exit code `0` after printing the scan result to stdout.
    """
    parser = argparse.ArgumentParser(description="Scan low-stock alerts and emit JSON.")
    parser.add_argument(
        "--db-path",
        default=str(DEFAULT_DB_PATH),
        help="SQLite database file to scan.",
    )
    parser.add_argument(
        "--days-ahead",
        type=int,
        default=LOW_STOCK_SCAN_DAYS,
        help="How many future days to scan for date_quantity alerts.",
    )
    args = parser.parse_args()

    with closing(ensure_database(args.db_path)) as conn:
        alerts = scan_low_stock_alerts(
            conn,
            mark_sent=True,
            days_ahead=args.days_ahead,
        )

    payload = {
        "status": "scanned",
        "reply": f"Detected {len(alerts)} low-stock alert(s).",
        "alerts": alerts,
    }
    print(json.dumps(payload, ensure_ascii=False, sort_keys=True))
    return 0


if __name__ == "__main__":
    raise SystemExit(main())
